src/org/sosy_lab/java_smt/solvers/princess/PrincessModel.java

```diff
@@ -240,14 +253,76 @@ public String toString() {
   @Override
   public void close() {}
 
+  /** Tries to determine the Sort of a Term. */
+  private Sort getSort(IExpression pTerm) {
+    // Just using sortof() won't work as Princess may have simplified the original term
+    // FIXME: This may also affect other parts of the code that use sortof()
```

Comment:
Would you be so kind and open an issue for this so that we don't forget?
Also, is there an option to disable this maybe?

Reply:
This is linked to https://github.com/sosy-lab/java-smt/pull/391#discussion_r1803154040. Once the Princess issue has been fixed we can simply use `Sort.sortOf`.